infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -129,10 +131,37 @@ public LogUnitServer(ServerContext serverContext) {
     /**
      * Service an incoming request for maximum global address the log unit server has written.
      */
-    @ServerHandler(type = CorfuMsgType.TAIL_REQUEST)
-    public void handleTailRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+    @ServerHandler(type = CorfuMsgType.LOG_TAIL_REQUEST)
+    public void handleLogTailRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+        Long logTail = batchWriter.queryLogTail(msg.getEpoch());
+        r.sendResponse(ctx, msg, CorfuMsgType.LOG_TAIL_RESPONSE.payloadMsg(logTail));
+    }
+
+    /**
+     * Service an incoming request for all stream tails (also returns log global tail).
+     */
+    @ServerHandler(type = CorfuMsgType.TAILS_REQUEST)
+    public void handleTailsRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
         TailsResponse tails = batchWriter.queryTails(msg.getEpoch());
-        r.sendResponse(ctx, msg, CorfuMsgType.TAIL_RESPONSE.payloadMsg(tails));
+        r.sendResponse(ctx, msg, CorfuMsgType.TAILS_RESPONSE.payloadMsg(tails));
+    }
+
+    /**
+     * Service an incoming request for streams addresses, i.e., the map of addresses belonging to each stream.
+     */
+    @ServerHandler(type = CorfuMsgType.ALL_STREAMS_ADDRESS_REQUEST)
+    public void handleStreamsAddressSpaceRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+        StreamsAddressResponse streamsAddressSpace = batchWriter.queryStreamsAddressSpace(msg.getEpoch());
```

Comment:
Why do we need to put this request into the queue and wait for the result? 
We can read the `address space` from streamLog and don't need BatchWriter for that.
Also, put the request to the queue brings some latency